mailbox/opensearch/src/test/java/org/apache/james/mailbox/opensearch/OpenSearchNoIndexBodyIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+                .host(tika.getIp())
+                .port(tika.getPort())
+                .timeoutInMillis(tika.getTimeoutInMillis())
+                .build()));
+        client = openSearch.getDockerOpenSearch().clientProvider().get();
+
+        InMemoryMessageId.Factory messageIdFactory = new InMemoryMessageId.Factory();
+        MailboxIdRoutingKeyFactory routingKeyFactory = new MailboxIdRoutingKeyFactory();
+        ReadAliasName readAliasName = new ReadAliasName(UUID.randomUUID().toString());
+        WriteAliasName writeAliasName = new WriteAliasName(UUID.randomUUID().toString());
+        indexName = new IndexName(UUID.randomUUID().toString());
+        MailboxIndexCreationUtil.prepareClient(client, readAliasName, writeAliasName, indexName,
+            openSearch.getDockerOpenSearch().configuration());
+
+        InMemoryIntegrationResources resources = InMemoryIntegrationResources.builder()
+            .preProvisionnedFakeAuthenticator()
+            .fakeAuthorizator()
+            .inVmEventBus()
+            .defaultAnnotationLimits()
+            .defaultMessageParser()
+            .listeningSearchIndex(preInstanciationStage -> new OpenSearchListeningMessageSearchIndex(
+                preInstanciationStage.getMapperFactory(),
+                ImmutableSet.of(),
+                new OpenSearchIndexer(client, writeAliasName),
+                new OpenSearchSearcher(client, new QueryConverter(new CriterionConverter()), SEARCH_SIZE, readAliasName, routingKeyFactory),
+                new MessageToOpenSearchJson(textExtractor, ZoneId.of("Europe/Paris"), IndexAttachments.YES, IndexHeaders.YES),
+                preInstanciationStage.getSessionProvider(), routingKeyFactory, messageIdFactory,
+                OpenSearchMailboxConfiguration.builder().indexBody(IndexBody.NO).build(), new RecordingMetricFactory()))
+            .noPreDeletionHooks()
+            .storeQuotaManager()
+            .build();
+
+        storeMailboxManager = resources.getMailboxManager();
+        messageSearchIndex = resources.getSearchIndex();
+
+        session = storeMailboxManager.createSystemSession(USERNAME);
+        inboxPath = MailboxPath.inbox(USERNAME);
+        storeMailboxManager.createMailbox(inboxPath, session);
+
+        StoreMessageManager inboxMessageManager = (StoreMessageManager) storeMailboxManager.getMailbox(inboxPath, session);
+        mailbox = inboxMessageManager.getMailboxEntity();
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        client.close();
+    }
+
+    @Test
+    void searchingByBodyContentShouldNotReturnMessageWhenNoIndexBody() throws Exception {
+        addMessage(session, inboxPath).block();
+
+        awaitForOpenSearch(QueryBuilders.matchAll().build()._toQuery(), 1L);
+
+        SearchQuery searchQuery = SearchQuery.of(SearchQuery.bodyContains("Hello"));
+
+        assertThat(messageSearchIndex.search(session, mailbox, searchQuery).collectList().block())
+            .isEmpty();
```

Comment:
Well fails:

```
java.lang.AssertionError: 
Expecting empty but was: [MessageUid{uid=1}]
```

Need to debug that, not sure where I messed up...

Reply:
Memory backend is not compliant with FetchTyle and always return the full message. CF `InMemoryMessageMapper::findInMailbox`.

We should cary other the indexed body config deep into `MimePartParser::parse` and use `message.getHeaderContent()` instead of `message.getFullContent()` in order to make it work.